core/pkg/errcapture/do.go

```diff
@@ -11,24 +11,36 @@ package errcapture
 import (
 	"io"
 	"io/ioutil"
+	"os"
 
 	"github.com/efficientgo/tools/core/pkg/merrors"
 	"github.com/pkg/errors"
 )
 
-type closerFunc func() error
+type doFunc func() error
 
-// Close runs function and on error return error by argument including the given error (usually
+// Do runs function and on error return error by argument including the given error (usually
 // from caller function).
-func Close(err *error, closer closerFunc, format string, a ...interface{}) {
-	*err = merrors.New(*err, errors.Wrapf(closer(), format, a...)).Err()
+func Do(err *error, doer doFunc, format string, a ...interface{}) {
+	derr := doer()
+	if err == nil {
+		return
+	}
+
+	// For os closers, it's a common case to double close. From reliability purpose this is not a problem it may only indicate
+	// surprising execution path.
+	if errors.Is(derr, os.ErrClosed) {
```

Comment:
It's weird that now these functions are kind of "generic" but they have these specific bits in them. Perhaps it should accept a slice of errors that would be ignored?

Reply:
Agree, is just a very common error that is breaking idempotency, so not sure 🤔